### PR TITLE
feat: Add Js Calculator Mobile First UI Design

### PR DESCRIPTION
**Pull request by moshoodabdullahi**
# This PR adds the Js Calculator Mobile First UI Design following the [Js Calculator Guide](https://github.com/bolah2009/command-line-guide/blob/master/guide/js/01-js-calculator-app.md) and design layout of the assignment link [TheOdinProject](https://www.theodinproject.com/paths/foundations/courses/foundations) and [Front-End Mentor](https://www.frontendmentor.io/challenges/calculator-app-9lteq5N29)
This Project Is built with:

- HTML5 elements
  - Semantic elements such as section, header and so on,

- CSS properties
   - CSS Positioning absolute, relative,
   - CSS Variables for reusable CSS values,
   - Typography such as font-size, font-weight and so on,
   - CSS Background for page aesthetic using background-color,
   - CSS flexbox such as justify-content, align-items and so on.
   - CSS Grid such as grid-column, grid-template-columns and so on.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # JS CALCULATOR
 
-> A Calculator webpage built using Js: addEventListener, querySelector, CSS Flexbox, Typography and Semantic HTML.
+> A Calculator webpage built using Js: addEventListener, querySelector, CSS Flexbox, CSS Grid, Typography and Semantic HTML.
```
Comment (teckcipher) [inline README.md]: CSS Variables are the major tech used in CSS

File: css/index.css [hunk 2]
```diff
@@ -2,3 +2,221 @@
...
+  --bg-result-toggle-key: hsl(6deg 63% 50%);
+  --key-shadow-result: hsl(6deg 70% 34%);
+  --text-color: hsl(30deg 25% 89%);
+  --text-del-result-color: hsl(0deg 0% 100%);
+  --text-result-color: hsl(221deg 14% 31%);
+  --equals-color: hsl(0deg 0% 100%);
+  --dot-position: 8px;
+  --del-reset-hover-color: #a2b3e1;
+  --key-hover: #fff;
+  --reset-dot-hover: #f96c5b;
```
Comment (teckcipher) [inline css/index.css]: Let's change this to `hsl` to make the color consistent

File: css/index.css [hunk 3]
```diff
@@ -2,3 +2,221 @@
...
+  --reset-dot-hover: #94fff9;
+}
+
+.dot {
+  background-color: var(--bg-result-toggle-key);
+  width: 15px;
+  height: 15px;
+  border-radius: 50%;
+  margin-left: 6px;
+  position: relative;
```
Comment (teckcipher) [inline css/index.css]: Let's make this a utility class

File: css/index.css [hunk 4]
```diff
@@ -2,3 +2,221 @@
...
+}
+
+.screen-input {
+  color: var(--text-color);
+}
+
+.header-text {
+  color: var(--text-color);
+  position: absolute;
```
Comment (teckcipher) [inline css/index.css]: This too - utility class

File: css/index.css [hunk 5]
```diff
@@ -2,3 +2,221 @@
...
+  --bg-theme: hsl(222deg 26% 31%);
+  --bg-toggle-keypad: hsl(223deg 31% 20%);
+  --bg-screen: hsl(224deg 36% 15%);
+  --bg-del-reset-key: hsl(225deg 21% 49%);
+  --key-shadow-del-reset: hsl(224deg 28% 35%);
+  --bg-keys: hsl(30deg 25% 89%);
+  --key-shadow-keys: hsl(28deg 16% 65%);
+  --bg-result-toggle-key: hsl(6deg 63% 50%);
+  --key-shadow-result: hsl(6deg 70% 34%);
+  --text-color: hsl(30deg 25% 89%);
```
Comment (teckcipher) [inline css/index.css]: Let's use more generic naming conventions, use `primary-color`, `secondary-color`, `additional-color`, `tertiary-color`, etc you can also append `text` or/and `bg`, e.g. `primary-text-color` or/and `primary-bg-color`

File: css/reset.css [hunk 6]
```diff
@@ -3,3 +3,22 @@
...
+
+*,
+*::after,
+*::before {
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+  font-family: Spartan, sans-serif;
+  font-size: 32px;
+}
```
Comment (teckcipher) [inline css/reset.css]: Let's merge this with the above css

File: css/reset.css [hunk 7]
```diff
@@ -3,3 +3,22 @@
...
+
+body {
+  height: 100vh;
+  background-color: var(--bg-theme);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
```
Comment (teckcipher) [inline css/reset.css]: Why is this needed for the body element? Can we move this to div or main element?

File: index.html [hunk 8]
```diff
@@ -6,10 +6,45 @@
...
+            <label class="theme-1-label" for="theme-1">1</label>
+
+            <label class="theme-2-label" for="theme-2">2</label>
+
+            <label class="theme-3-label" for="theme-3">3</label>
+          </div>
+        </div>
+      </header>
+      <div class="screen d-flex">
+        <div class="screen-input">399,981</div>
```
Comment (teckcipher) [inline index.html]: I dont think `input` is he right word here, perhaps `display`?
I would recommend using an id attribute as well so it can be used to target the element with js

File: index.html [hunk 9]
```diff
@@ -6,10 +6,45 @@
...
+        <div class="theme-container">
+          <div class="theme-input">
+            <div class="dot"></div>
+            <label class="theme-1-label" for="theme-1">1</label>
+
+            <label class="theme-2-label" for="theme-2">2</label>
+
+            <label class="theme-3-label" for="theme-3">3</label>
+          </div>
+        </div>
```
Comment (teckcipher) [inline index.html]: This should be a radio button, you can learn from @Yinkusadel

File: index.html [hunk 10]
```diff
@@ -6,10 +6,45 @@
...
+        <button class="key" value="2">2</button>
+        <button class="key" value="3">3</button>
+        <button class="operator" value="-">-</button>
+        <button class="key" value=".">.</button>
+        <button class="key" value="0">0</button>
+        <button class="operator" value="/">/</button>
+        <button class="operator" value="x">x</button>
+        <button class="reset" value="reset" id="reset-key">RESET</button>
+        <button class="equals-key equal" value="=" id="equals-key">=</button>
+      </div>
```
Comment (teckcipher) [inline index.html]: Nice work with the button arrangements.

1. Please all buttons should have a value attribute, it would be used in your js
2. All buttons should have a type attribute `button`
3. All buttons should have a general class `key`, then an additional class for special cases like you are already doing
4. Let's avoid the use of `id` for these buttons, the value should be enough for all the features needed.

File: index.html [hunk 11]
```diff
@@ -6,10 +6,45 @@
...
     </main>
-    <script src="js/index.js"></script>
```
Comment (teckcipher) [inline index.html:13]: Any reason why this is removed?

**Review by teckcipher — CHANGES_REQUESTED** [review]

File: index.html (hunk 11 above)
Comment (moshoodabdullahi) [reply index.html:13]: It was a mistake sir, and also the innerHTML in the JS file was overriding the UI and only displaying Hello, World

File: css/index.css (hunk 2 above)
Comment (moshoodabdullahi) [reply css/index.css]: Done

File: css/reset.css (hunk 6 above)
Comment (moshoodabdullahi) [reply css/reset.css]: I have merge this sir

File: index.html (hunk 8 above)
Comment (moshoodabdullahi) [reply index.html]: I have changed the input to display sir

**Comment by Yinkusadel** [comment]
Please update the TOP assignment link, and screenshot of your design

File: css/index.css [hunk 12]
```diff
@@ -2,3 +2,248 @@
...
+  --key-hover: hsl(268deg 54% 44%);
+  --reset-dot-hover: hsl(177deg 100% 79%);
+}
+
+.input {
+  height: 30px;
+  width: 80px;
+  border-radius: 30px;
+  background-color: var(--bg-toggle-keypad);
+  position: relative;
```
Comment (Yinkusadel) [inline css/index.css]: Can you please use the utility class for this

File: css/index.css [hunk 13]
```diff
@@ -2,3 +2,248 @@
...
+  left: 90px;
+}
+
+.theme-choice-1,
+.theme-choice-2,
+.theme-choice-3 {
+  appearance: none;
+  position: absolute;
```
Comment (Yinkusadel) [inline css/index.css]: and this

**Review by Yinkusadel — CHANGES_REQUESTED** [review]
Please do that and you're good

File: css/index.css (hunk 12 above)
Comment (moshoodabdullahi) [reply css/index.css]: I have done this

File: css/index.css (hunk 13 above)
Comment (moshoodabdullahi) [reply css/index.css]: I have done this too

**Review by Yinkusadel — DISMISSED** [review]
Fine by me 👍

**Review by teckcipher — APPROVED** [review]

**Review by Yinkusadel — APPROVED** [review]
good to go
